frontend/src/components/game/Join.jsx

```diff
@@ -6,13 +6,58 @@ import ThreeScene from '../game/Game';
 
 export default function Join({ remoteGame }) {
 
-	const [gameSettings, setGameSettings] = useState(() => {
-		const fetchedSettings = localStorage.getItem('gameSettings');
-		const parsedSettings = JSON.parse(fetchedSettings);
-		console.log('fetched settings: ' + fetchedSettings);
-		return parsedSettings || "";
+	const { data: session, status } = useSession();
+	const [userId, setUserId] = useState(null);
+	const [gameSettings, setGameSettings] = useState({
```

Comment:
So you have a useEffect call that sets the value of gameSettings. You don't need to set these values beforehand.

Also, setting userId to -1 in this state isn't required. Let it come from the userId state. Don't replicate the same variable in two places in the same file. It can cause confusion and further problems.